qt/widgets/plotting/src/Mpl/ExternalPlotter.cpp

```diff
@@ -132,17 +142,35 @@ void ExternalPlotter::plotSpectra(std::string const &workspaceName, std::string
  * @param workspaceIndices List of indices to plot
  */
 void ExternalPlotter::plotCorrespondingSpectra(std::vector<std::string> const &workspaceNames,
-                                               std::vector<int> const &workspaceIndices, bool errorBars) {
+                                               std::vector<int> const &workspaceIndices, bool errorBars,
+                                               boost::optional<std::vector<QHash<QString, QVariant>>> kwargs,
+                                               boost::optional<std::vector<bool>> errorBarsPerSpectra) {
   if (workspaceNames.empty() || workspaceIndices.empty())
     return;
   if (workspaceNames.size() > 1 && workspaceNames.size() != workspaceIndices.size())
     return;
-
-  auto figure = workbenchPlot(QStringList(QString::fromStdString(workspaceNames[0])), {workspaceIndices[0]}, errorBars);
-  for (auto i = 1u; i < workspaceNames.size(); ++i) {
-    if (figure)
-      figure = workbenchPlot(QStringList(QString::fromStdString(workspaceNames[i])), {workspaceIndices[i]}, errorBars,
-                             boost::none, figure.get());
+  if (!errorBarsPerSpectra)
+    errorBarsPerSpectra = std::vector<bool>(workspaceIndices.size(), errorBars);
+  else if (errorBarsPerSpectra->size() != workspaceIndices.size())
+    return;
+  if (kwargs) {
```

Comment:
It would be nice to avoid this divergence and to just call the same piece of code without worrying whats in the kwargs. I think you could do this by changing the kwargs parameter such that you pass in the following to the function
`std::vector<boost::optional<QHash<QString, QVariant>>> kwargs`

I am not sure if that was a clear explanation so let me know if it didn't make sense